include/gatekeeper_net.h

```diff
@@ -19,8 +19,19 @@
 #ifndef _GATEKEEPER_NET_H_
 #define _GATEKEEPER_NET_H_
 
+#include <stdint.h>
+
+/* Configuration for the Network. */
+struct net_config {
+	uint16_t	num_rx_queues;
+	uint16_t	num_tx_queues;
+	uint32_t	num_ports;
+};
+
+struct net_config *alloc_and_return_net_conf(void);
+
 int gatekeeper_init_network(void);
 
-void gatekeeper_free_network(void);
+int gatekeeper_free_network(void);
```

Comment:
There's no need to return `int`, keep `void`. What this function need is to receive `struct net_config` to avoid a global variable.